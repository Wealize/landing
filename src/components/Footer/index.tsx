import React from 'react'
import Link from 'next/link'

import WealizeWhiteNormalMediumIcon from '../icons/WealizeWhiteNormalMediumIcon'
import ButtonRounded from '../Buttons/Rounded/'
import {
  Container,
  ContainerContent,
  ContainerGetInTouch,
  ContainerLogo,
  BigText,
  ContainerWorkWithUs,
  ContainerBottom,
  Divider,
  ContainerSocial,
  Social,
  PrivacyLegalLink,
  CopyrightText,
  ContainerGrants,
  Grant
} from './styles'
import { ACCENT, DARK_GRAY, WHITE } from '../../theme/color'
import LinkedinIcon from '../icons/social/LinkedinIcon'
import TwitterIcon from '../icons/social/TwitterIcon'
import DevtoIcon from '../icons/social/DevtoIcon'
import CDTIIcon from '../icons/grants/CDTIIcon'
import InnovationPymeIcon from '../icons/grants/InnovationPymeIcon'
const Footer = () => {
  return (
    <Container>
      <ContainerLogo>
        <WealizeWhiteNormalMediumIcon />
      </ContainerLogo>
      <ContainerContent>
        <ContainerGetInTouch>
          <BigText>
            Contact us about your project, your idea or for more information
            about our Work
          </BigText>
          <ButtonRounded
            backgroundColor={DARK_GRAY}
            textColor={WHITE}
            accentColor={ACCENT}
          >
            Get in touch
          </ButtonRounded>
        </ContainerGetInTouch>
        <ContainerWorkWithUs>
          <BigText>
            Join the culture. We are always looking for talent to grow our
            global community
          </BigText>
          <ButtonRounded
            backgroundColor={DARK_GRAY}
            textColor={WHITE}
            accentColor={ACCENT}
          >
            Work with us
          </ButtonRounded>
        </ContainerWorkWithUs>
      </ContainerContent>
      <ContainerBottom>
        <Divider></Divider>
        <ContainerSocial>
          <Social>
            <LinkedinIcon />
          </Social>
          <Social>
            <TwitterIcon />
          </Social>
          <Social>
            <DevtoIcon />
          </Social>
        </ContainerSocial>
        <Link href="">
          <PrivacyLegalLink>Privacy & Legal</PrivacyLegalLink>
        </Link>
        <CopyrightText>Wealize 2020. All Rights Reserved.</CopyrightText>
        <Divider></Divider>
        <ContainerGrants>
          <Grant>
            <CDTIIcon />
          </Grant>
          <Grant>
            <InnovationPymeIcon />
          </Grant>
        </ContainerGrants>
      </ContainerBottom>
    </Container>
  )
}

export default React.memo(Footer)
