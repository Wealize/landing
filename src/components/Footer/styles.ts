import styled from 'styled-components'

import { WHITE_COLOR, DARK_GRAY_COLOR } from '../../theme/color'
import {
  sixthPx,
  singlePx,
  doublePx,
  triplePx,
  singleAndHalfPx,
  quadruplePx
} from '../../theme/space'
import { media } from '../../theme/media'
import { CONTAINER_SPACING } from '../../theme/spacingGuide'

export const Container = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  background-color: ${DARK_GRAY_COLOR};
  ${CONTAINER_SPACING.MOBILE}

  ${media.greaterThan('md')`
    ${CONTAINER_SPACING.TABLET}
  `};

  ${media.greaterThan('lg')`
    ${CONTAINER_SPACING.SMALL_DESKTOP};
  `};

  ${media.greaterThan('xl')`
    ${CONTAINER_SPACING.DESKTOP};
  `};

  ${media.greaterThan('xxl')`
    ${CONTAINER_SPACING.WIDE};
  `};
`

export const ContainerContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;

  ${media.greaterThan('md')`
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: ${singlePx()};
  `}

  ${media.greaterThan('xl')`
    grid-gap: ${doublePx()};
  `}

  ${media.greaterThan('xxl')`
    grid-gap: ${quadruplePx()};
  `}
`

export const ContainerLogo = styled.div`
  display: flex;
  width: 9rem;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding-bottom: ${singlePx()};

  ${media.greaterThan('md')`
    width: 8rem;
    padding-bottom: ${singleAndHalfPx()};
  `};

  ${media.greaterThan('lg')`
    padding-bottom: ${doublePx()};
  `};
`

export const BigText = styled.h5`
  color: ${WHITE_COLOR};
  padding: ${singlePx()} 0;
  font-size: 1.8rem;
  line-height: 1.33;

  ${media.greaterThan('lg')`
    font-size: 2.5rem;
  `};
`

export const ContainerContactSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  padding: ${doublePx()} 0;

  > .button-rounded {
    margin-top: 5%;
    min-width: 270px;

    & span {
      font-weight: normal;
    }
  }

  ${media.greaterThan('md')`
    padding: 0;
    justify-content: space-between;
    min-height: 350px;

    > .button-rounded {
      margin-top: 0;
      min-width: 250px;
    }
  `};

  ${media.greaterThan('lg')`
    min-height: 450px;
  `};

  ${media.greaterThan('xl')`
    min-height: 400px;
  `};
`

export const DividerMobile = styled.hr`
  width: 100%;
  border: 1px solid ${WHITE_COLOR};
  margin: ${doublePx()} 0;

  ${media.greaterThan('md')`
    display: none;
  `};
`

export const Divider = styled.hr`
  width: 100%;
  border: 1px solid ${WHITE_COLOR};
  margin: ${doublePx()} 0;

  ${media.greaterThan('md')`
    margin: ${singlePx()} 0;
  `};
`

export const ContainerBottom = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

export const ContainerSocialMobile = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  ${media.greaterThan('md')`
    display: none;
  `};
`

export const ContainerPrivacyCopyright = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  ${media.greaterThan('md')`
    width: 100%;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    margin-top: ${triplePx()};
  `};
`

export const ContainerSocial = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  display: none;

  ${media.greaterThan('md')`
    display: flex;
    width: 30%;
    justify-content: flex-end;
  `};

  ${media.greaterThan('lg')`
    width: 20%;
  `};

  ${media.greaterThan('xl')`
    width: 15%;
  `};
`

export const Social = styled.a`
  width: 46px;
  height: 46px;
  margin: 0 ${sixthPx()};
  text-decoration: none;

  ${media.greaterThan('md')`
    width: 56px;
    height: 56px;
  `};

  ${media.greaterThan('lg')`
    width: 60px;
    height: 66px;
  `};

  ${media.greaterThan('xl')`
    width: 66px;
    height: 66px;
  `};
`

export const PrivacyLegalLink = styled.a`
  color: ${WHITE_COLOR};
  font-size: 1.1rem;
  padding: ${singlePx()} 0;
  text-decoration: underline;

  ${media.greaterThan('md')`
    padding:  0;
    margin-right: 10%;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.5rem;
  `};
`

export const CopyrightText = styled.span`
  color: ${WHITE_COLOR};
  font-size: 1.1rem;

  ${media.greaterThan('lg')`
    font-size: 1.5rem;
  `};
`

export const ContainerGrantsAndSocials = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;

  ${media.greaterThan('md')`
    justify-content: space-between;
  `};

  ${media.greaterThan('xl')`
    padding-top: ${doublePx()};
  `};
`

export const ContainerGrants = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;

  ${media.greaterThan('md')`
    width: 70%;
    justify-content: flex-start;
  `};

  ${media.greaterThan('lg')`
    width: 80%;
  `};

  ${media.greaterThan('xl')`
    width: 85%;
  `};
`

export const Grant = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: ${singlePx()};

  & a {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  ${media.greaterThan('md')`
    width: 50%;
    margin-bottom: 0;
  `};

  ${media.greaterThan('lg')`
    width: 40%;
  `};

  ${media.greaterThan('xl')`
    width: 25%;
  `};
`
