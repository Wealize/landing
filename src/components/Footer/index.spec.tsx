import React from 'react'
import { shallow } from 'enzyme'

import ButtonRounded from '../Buttons/Rounded'
import WealizeWhiteNormalMediumIcon from '../icons/WealizeWhiteNormalMediumIcon'

import Footer from './index'
import {
  BigText,
  Social,
  InlineLink,
  CopyrightText,
  Grant
} from './styles'


describe('<Footer />', () => {
  const wrapper = shallow(<Footer />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('wrapper: render content correctly', () => {
    expect(wrapper.find(WealizeWhiteNormalMediumIcon).length).toEqual(1)
    expect(wrapper.find(BigText).length).toEqual(2)
    expect(wrapper.find(ButtonRounded).length).toEqual(2)
    expect(wrapper.find(InlineLink).length).toEqual(3)
    expect(wrapper.find(CopyrightText).length).toEqual(1)
    expect(wrapper.find(Social).length).toEqual(6)
    expect(wrapper.find(Grant).length).toEqual(2)
  })
})
