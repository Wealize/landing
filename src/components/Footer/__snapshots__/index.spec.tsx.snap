// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> wrapper: should render 1`] = `
<styles__Container
  id="footer"
>
  <styles__ContainerLogo>
    <Memo(WealizeWhiteNormalMediumIcon) />
  </styles__ContainerLogo>
  <styles__ContainerContent>
    <styles__ContainerContactSection>
      <styles__BigText>
        common:footer.get_in_touch.text
      </styles__BigText>
      <Memo(ButtonRounded)
        animatedBackgroundColor="#6666ff"
        animatedTextColor="#ffffff"
        backgroundColor="#3c3c3b"
        handleClick={[Function]}
        id="footer-get-in-touch-button"
        size={
          Object {
            "desktop": Array [
              200,
              85,
            ],
            "mobile": Array [
              180,
              75,
            ],
            "tablet": Array [
              180,
              80,
            ],
            "wide": Array [
              220,
              90,
            ],
          }
        }
        textColor="#ffffff"
      >
        common:footer.get_in_touch.button_text
      </Memo(ButtonRounded)>
    </styles__ContainerContactSection>
    <styles__ContainerContactSection>
      <styles__BigText>
        common:footer.work_with_us.text
      </styles__BigText>
      <Memo(ButtonRounded)
        animatedBackgroundColor="#6666ff"
        animatedTextColor="#ffffff"
        backgroundColor="#3c3c3b"
        handleClick={[Function]}
        id="footer-work-whit-us-button"
        size={
          Object {
            "desktop": Array [
              200,
              85,
            ],
            "mobile": Array [
              180,
              75,
            ],
            "tablet": Array [
              180,
              80,
            ],
            "wide": Array [
              220,
              90,
            ],
          }
        }
        textColor="#ffffff"
      >
        common:footer.work_with_us.button_text
      </Memo(ButtonRounded)>
    </styles__ContainerContactSection>
  </styles__ContainerContent>
  <styles__ContainerBottom>
    <styles__DividerMobile />
    <styles__ContainerSocialMobile>
      <Link
        href="https://www.linkedin.com/company/wealizedigital/"
        passHref={true}
      >
        <styles__Social
          aria-label="Linkedin"
          rel="noreferrer"
          target="_blank"
          title="Wealize Linkedin link"
        >
          <LinkedinIcon />
        </styles__Social>
      </Link>
      <Link
        href="https://twitter.com/wealizedigital"
        passHref={true}
      >
        <styles__Social
          aria-label="Twitter"
          rel="noreferrer"
          target="_blank"
          title="Wealize Twitter link"
        >
          <TwitterIcon />
        </styles__Social>
      </Link>
      <Link
        href="https://dev.to/wealize"
        passHref={true}
      >
        <styles__Social
          aria-label="Dev.to"
          rel="noreferrer"
          target="_blank"
          title="Wealize Dev.to link"
        >
          <DevtoIcon />
        </styles__Social>
      </Link>
    </styles__ContainerSocialMobile>
    <styles__ContainerPrivacyCopyright>
      <Link
        href="/privacy-legal"
        passHref={true}
      >
        <styles__PrivacyLegalLink
          aria-label="Privacy & Legal"
          data-cy="footer-privacy-link"
          title="Privacy page link"
        >
          common:footer.privacy_legal
        </styles__PrivacyLegalLink>
      </Link>
      <styles__CopyrightText>
        common:footer.copyright
      </styles__CopyrightText>
    </styles__ContainerPrivacyCopyright>
    <styles__Divider />
    <styles__ContainerGrantsAndSocials>
      <styles__ContainerGrants>
        <styles__Grant>
          <Link
            href="/grants/cdti"
            passHref={true}
          >
            <a
              aria-label="CDTI"
              data-cy="footer-cdti-link"
              title="CDTI page link"
            >
              <Memo(CDTIIcon) />
            </a>
          </Link>
        </styles__Grant>
        <styles__Grant>
          <Link
            href="/grants/innovative-sme"
            passHref={true}
          >
            <a
              aria-label="Innovation Pyme"
              data-cy="footer-innovation-pyme-link"
              title="Innovation Pyme page link"
            >
              <Memo(InnovationPymeIcon) />
            </a>
          </Link>
        </styles__Grant>
      </styles__ContainerGrants>
      <styles__ContainerSocial>
        <Link
          href="https://www.linkedin.com/company/wealizedigital/"
          passHref={true}
        >
          <styles__Social
            rel="noreferrer"
            target="_blank"
          >
            <LinkedinIcon />
          </styles__Social>
        </Link>
        <Link
          href="https://twitter.com/wealizedigital"
          passHref={true}
        >
          <styles__Social
            rel="noreferrer"
            target="_blank"
          >
            <TwitterIcon />
          </styles__Social>
        </Link>
        <Link
          href="https://dev.to/wealize"
          passHref={true}
        >
          <styles__Social
            rel="noreferrer"
            target="_blank"
          >
            <DevtoIcon />
          </styles__Social>
        </Link>
      </styles__ContainerSocial>
    </styles__ContainerGrantsAndSocials>
  </styles__ContainerBottom>
</styles__Container>
`;
